### PR TITLE
mpl lightness test

### PR DESCRIPTION
**Pull request by samjrholt**

### CODE REVIEW

File: discretisedfield/tests/test_field.py [hunk 1]
```diff
@@ -3029,6 +3029,22 @@ def test_mpl_lightess(test_field):
...
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_mpl_dimension_lightness(valid_mesh, nvdim):
+    field = df.Field(valid_mesh, nvdim=nvdim)
+
+    if valid_mesh.region.ndim != 2:
+        with pytest.raises(RuntimeError):
+            field.mpl.lightness()
+    elif nvdim not in [1, 2]:
+        with pytest.raises(ValueError):
+            field.mpl.lightness()
```
Comment (lang-m) [inline discretisedfield/tests/test_field.py:3041]: Lightness also works for `nvdim=3`. Both 2 and 3 vdims require `vdim_mapping` to be defined. I think we should modify the implementation and raise a `RuntimeError` when `nvdim > 3`.

File: discretisedfield/tests/test_field.py [hunk 2]
```diff
@@ -3033,12 +3033,12 @@ def test_mpl_lightess(test_field):
...
     field = df.Field(valid_mesh, nvdim=nvdim)
 
-    if valid_mesh.region.ndim != 2:
+    if valid_mesh.region.ndim != 2 or nvdim > 3:
         with pytest.raises(RuntimeError):
             field.mpl.lightness()
-    elif nvdim not in [1, 2]:
-        with pytest.raises(ValueError):
-            field.mpl.lightness()
+    elif nvdim == 3:
```
[the remark this answers is not in the document]
Comment (samjrholt) [reply discretisedfield/tests/test_field.py:3039]: As it's a plotting method everything that gets to this stage is already a 2D plane

[the remark this answers is not in the document]
Comment (lang-m) [reply discretisedfield/tests/test_field.py:3039]: That is correct. But: If `nvdim` is 3 and our input mesh (`valid_mesh`) has `ndim=3` we create a field (in l. 3034) with a meaningful `vdim_mapping` that should stay meaningful when we select a single plane. We should test for this and hence only set the `vdim_mapping` manually if required.

[the remark this answers is not in the document]
Comment (samjrholt) [reply discretisedfield/tests/test_field.py:3039]: This test is not designed to select a plane from a higher order mesh. We are just checking the range of valid meshes, hence, if we want to plot anything that is not `ndim==2` we need to set `vdim_mapping` manually. 

`we create a field (in l. 3034) with a meaningful vdim_mapping that should stay meaningful when we select a single plane` I agree with this statement but I don't think this is a plotting test. This is testing the `sel` method works properly.

[the remark this answers is not in the document]
Comment (samjrholt) [reply discretisedfield/tests/test_field.py:3039]: We also have the `test_mpl_vector` function (l. 3032) which does start with a 3D test field and uses sel.

[the remark this answers is not in the document]
Comment (lang-m) [reply discretisedfield/tests/test_field.py:3039]: You are right. I did not keep in mind that we do not use `sel` in this test.

**Review by lang-m — APPROVED** [review]

**Pull request merged**